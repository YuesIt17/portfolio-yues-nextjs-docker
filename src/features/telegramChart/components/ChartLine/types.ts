export type TChartLine = {
  points: string;
  fill?: string;
  stroke?: string;
  isVisible?: boolean;
};
