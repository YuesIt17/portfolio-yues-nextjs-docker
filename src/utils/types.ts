import {TColumnsValueAxis} from '../api/telegramChart/types';

export type TColumnValue = {
  values: TColumnsValueAxis;
  type: string;
  name: string;
  color: string;
};

export type TChartDataLabel = {
  label: string;
  coordinate: number;
};

export type TChartDataLine = {
  type: string;
  name: string;
  color: string;
  points: string;
  pointsMap: string;
  isVisible: boolean;
};

export type TChartPrepareData = {
  lines: TChartDataLine[];
  maxDataX: number;
  maxDataY: number;
  labelsX?: TChartDataLabel[];
  labelsY?: TChartDataLabel[];
};

export type TCoordinatePoint = {
  x: number;
  y: number;
};
