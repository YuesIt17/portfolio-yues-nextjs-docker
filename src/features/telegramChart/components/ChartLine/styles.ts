import {css, Theme} from '@emotion/react';
import {TChartLine} from './types';

export const useStyles = ({
  fill,
  stroke,
}: Pick<TChartLine, 'fill' | 'stroke'>) => ({
  line: (theme: Theme) =>
    css({
      fill: fill || 'none',
      stroke: stroke || theme.colors.primary.main,
    }),
});
