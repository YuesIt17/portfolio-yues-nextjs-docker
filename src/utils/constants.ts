export const DELTA_DATE_TIME_TO_INT = 5;
