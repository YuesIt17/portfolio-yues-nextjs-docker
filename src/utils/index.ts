export {prepareData} from './prepareData';
export {
  ThemeTestProvider,
  authTestRender,
  themeTestRender,
  authWrapper,
} from './tests';
