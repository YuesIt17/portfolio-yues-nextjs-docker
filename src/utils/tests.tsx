import {AuthProvider} from '@/components';
import {ThemeProvider} from '@emotion/react';
import {render, RenderOptions, RenderResult} from '@testing-library/react';
import React, {FC, ReactElement, ReactNode} from 'react';
import {theme} from '../app/App';
import renderer, {
  ReactTestRenderer,
  TestRendererOptions,
} from 'react-test-renderer';

export const ThemeTestProvider: FC = ({children}) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export const themeTestRender = (
  ui: ReactElement,
  options?: RenderOptions
): RenderResult => {
  return render(<ThemeTestProvider>{ui}</ThemeTestProvider>, options);
};

export const themeTestRenderer = (
  ui: ReactElement,
  options?: TestRendererOptions
): ReactTestRenderer => {
  return renderer.create(<ThemeTestProvider>{ui}</ThemeTestProvider>, options);
};

export const authTestRender = (
  ui: ReactElement,
  options?: RenderOptions
): RenderResult => {
  return render(
    <AuthProvider>
      <ThemeTestProvider>{ui}</ThemeTestProvider>
    </AuthProvider>,
    options
  );
};

export const authWrapper = ({children}: {children?: ReactNode | undefined}) => (
  <AuthProvider>{children}</AuthProvider>
);
