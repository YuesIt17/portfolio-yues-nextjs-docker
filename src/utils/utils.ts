import {flowRight, join} from 'lodash';

export const joinValues = (values: string[], separator = ' ') =>
  join(values, separator);

export function parseDateTimeToInt(
  value: number,
  index: number,
  delta: number
) {
  const countDigit = value.toString().length;
  const rank = new Array(countDigit).fill(1).join('');
  return Math.round(Number(value) / Number(rank) + index * delta);
}

export const compose = flowRight;
