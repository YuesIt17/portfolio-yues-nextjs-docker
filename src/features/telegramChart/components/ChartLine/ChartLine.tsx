import React from 'react';
import {useStyles} from './styles';
import {TChartLine} from './types';

export const ChartLine = ({
  points,
  fill,
  stroke,
  isVisible = true,
}: TChartLine) => {
  const styles = useStyles({fill, stroke});

  if (!isVisible) return null;
  return <polyline css={styles.line} points={points} />;
};
