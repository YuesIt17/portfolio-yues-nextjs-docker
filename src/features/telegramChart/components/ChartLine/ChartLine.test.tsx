import React from 'react';
import {themeTestRenderer} from '@/utils/tests';
import {ChartLine} from './ChartLine';

describe('ChartLine test', () => {
  test('Check render ChartLine component', () => {
    const chartAxisX = themeTestRenderer(
      <ChartLine points="1,1 2,2" />
    ).toJSON();

    expect(chartAxisX).toMatchSnapshot();
  });
});
