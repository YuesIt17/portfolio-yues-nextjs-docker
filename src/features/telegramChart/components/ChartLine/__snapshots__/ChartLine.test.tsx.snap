// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartLine test Check render ChartLine component 1`] = `
<polyline
  className="css-1nfezsp-line"
  points="1,1 2,2"
/>
`;
